From f9d6627fe03f68a132501a488f73e08080018115 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 3 May 2019 16:31:11 +0200
Subject: [PATCH] backends: Fallback to builtin panel for devices where all
 heuristics fail

This is 1) relatively likely as not all touchscreens are nice enough to
report a device size that will help us here and 2) Better than nothing if
everything fails anyway, as it will break on multi-monitor and non-default
monitor rotations.

Closes: https://gitlab.gnome.org/GNOME/mutter/issues/581
---
 src/backends/meta-input-mapper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/meta-input-mapper.c b/src/backends/meta-input-mapper.c
index 5227292dc..acc9b1618 100644
--- a/src/backends/meta-input-mapper.c
+++ b/src/backends/meta-input-mapper.c
@@ -381,7 +381,7 @@ guess_candidates (MetaInputMapper     *mapper,
       info->candidates[META_MATCH_SIZE] = matched_monitor;
     }
 
-  if (input->builtin)
+  if (input->builtin || best == N_OUTPUT_MATCHES)
     {
       best = MIN (best, META_MATCH_IS_BUILTIN);
       find_builtin_output (mapper, &info->candidates[META_MATCH_IS_BUILTIN]);
-- 
2.18.1

